src/System.Collections.Concurrent/tests/ProducerConsumerCollectionTests.cs

```diff
@@ -761,8 +768,11 @@ public void ManyConcurrentAddsTakes_CollectionRemainsConsistent(int operations)
                 // avoid 0 as default(T), to detect accidentally reading a default value
                 for (int i = 1; i < (1 + operations); i++)
                 {
-                    Assert.True(c.TryAdd(i));
-                    added.Add(i);
+                    if (c.Count < MaxConcurrentCollectionSize)
+                    {
+                        Assert.True(c.TryAdd(i));
+                        added.Add(i);
+                    }
```

Comment:
Updated. The upper max on collection size is now 9 million when the starts align.